views/pulsar/v2/helpers/elem.html.twig

```diff
@@ -260,7 +260,7 @@ value       | string | Specifies the value of the input
                     {% if option is iterable %}
                         {% for item in option %}
                             <option {{ util.value(item.value) }}
-                                {%- if mainOptions.selected is defined and item.value == mainOptions.selected %} selected{% endif -%}
+                                {%- if mainOptions.selected is defined and item.value in mainOptions.selected %} selected{% endif -%}
```

Comment:
If we're changing the selected option to accept both strings and arrays, will `in` partially match strings here and potentially select values we don't want?

So `'foo' in ['foo', 'bar'] === 'foo'` whereas `'foo' in 'foobar' === 'foo'` ?

Reply:
The docs do make it sound that way, but it didn't work in my tests.

**Given** 

```
{{
    form.select({
        'selected': ['foo', 'ba'],
        'options': [
            {
                'label': 'bar',
                'value': 'foo'
            },
            {
                'label': 'baz',
                'value': 'bar'
            }
        ]
    })
}}
```

**Then**

```
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'<div class="form__group"><div class="controls"><select class="form__control select"><option value="foo" selected>bar</option><option value="bar" selected>baz</option></select></div></div>'
+'<div class="form__group"><div class="controls"><select class="form__control select"><option value="foo" selected>bar</option><option value="bar">baz</option></select></div></div>'
```

Reply:
I'll add test cases to test this behaviour, in case the twig internals change at some point

Reply:
your test there passes an array in the selected option, the issue should rear it's head if you changed your test to:

```{{
    form.select({
        'selected': 'ba',
        'options': [
            {
                'label': 'bar',
                'value': 'foo'
            },
            {
                'label': 'baz',
                'value': 'bar'
            }
        ]
    })
}}```

Reply:
I did, see this test case: 
https://github.com/jadu/pulsar/pull/906/commits/37fa82f5b627026d780f9acd56648ddd56447822#diff-0bc22e4053d041ee111cb73f1013690b

Reply:
To me it feels like we should match the documented behaviour and just perform a check on the type of the selected option (or limit the component to only accept 1 type). If somebody updated their twig version, the Pulsar unit tests wouldn't help us there would it?

Reply:
Sure, I see your point, happy to go this way

Reply:
Pushed a change which is more explicit in testing whether `selected` is a string/array.